src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -19,7 +20,7 @@ namespace System.ComponentModel.DataAnnotations
     /// </remarks>
     internal sealed class ValidationAttributeStore
     {
-        private readonly Dictionary<Type, TypeStoreItem> _typeStoreItems = new Dictionary<Type, TypeStoreItem>();
+        private readonly ConcurrentDictionary<Type, TypeStoreItem> _typeStoreItems = new();
```

Comment:
ValidationAttributeStore is used as a singleton so there's really only one of these, right?

Reply:
Yes, looks other calling code is always using `ValidationAttributeStore.Instance`.

Reply:
Yup